dice-game-ui-angular/src/app/services/local-game-storage-service.ts

```diff
@@ -1,33 +1,70 @@
 import { Injectable } from '@angular/core';
 import { Guid } from 'guid-typescript';
-import { GameEngine, SimpleGameEngine, Size } from 'engine';
-import { GameStorageService } from './game-storage-service';
+import { GameEngine, SimpleGameEngine } from 'engine';
+import { GameStorageService, SavedGameEngine } from './game-storage-service';
 
 @Injectable()
 export class LocalGameStorageService implements GameStorageService {
+  private readonly savedGamesKey = 'saved-games';
+
   createGame(gameEngine: GameEngine): Guid {
```

Comment:
I think you can remove `createGame` method and merge the logic with `saveGame`. It doesn't have much sense to separate these concepts.

Reply:
Actually, I did this to evade passing a gameEngine object as a parameter to the function. I decided that passing 'null' is not a good idea while creating a new game. If you don't agree with that, I'll change it.

Reply:
Maybe you misprinted and you meant `Guid` object, because you already passes game engine to both methods.

Reply:
Yes, of course, I meant guid object.

Reply:
You can add a default value to guid param of `saveGame` function which will equal to `null`. And then you will call `saveGame(engine)` from start game component and `saveGame(engine, engine.guid)` in game component.

Reply:
Moved creating to the saveGame